src/util/color/predefinedcolormap.h

```diff
@@ -0,0 +1,39 @@
+#ifndef PREDEFINEDCOLORMAP_H
+#define PREDEFINEDCOLORMAP_H
+
+#include <QColor>
+#include <QHash>
+
+#include "util/color/predefinedcolor.h"
+
+// Maps a PredefinedColorMap to an actual QColor (a.k.a its representation).
+//
+// Initially no color has a representation set.
+// Call setRepresentation(PredefinedColorPointer, QColor) to add representations and customize the color map.
+//
+// Uses the color's name() property as key, e.g. "#A9A9A9"
+// Since QHash has copy-on-write, making a copy of ColorsRepresentation is fast.
+// A deep copy of the QHash will be made when a copy is modified.
+class PredefinedColorMap final {
+  public:
+    // Set a color representation for a given color
+    void setRepresentation(PredefinedColorPointer color, QColor representation) {
+        m_colorNameMap[color->m_defaultRepresentation.name()] = representation.name();
```

Comment:
Do we have only one customized color per original color? 
How about just add a cutomizedRgba field to each PredefinedColor?

Reply:
> Do we have only one customized color per original color?

Yes, one or none.

> How about just add a cutomizedRgba field to each PredefinedColor?

Because the cutomizedRgba can change on skin reload, we would need the predefined colors to be mutable global state. With this map class, we can keep the predefined colors immutable.

Reply:
> How about just add a cutomizedRgba field to each PredefinedColor?

We might want to represent a color with different Rgba simultaneously. With the current implementation, you can customize the cue colors per waveform. Supporting this with mutable color instances seems complicated to me.

The mutable part is isolated from the immutable part, and the immutable part is trivially thread-safe. Then, clients just get a copy of the default color map, and modify it at will. Thus, mutation is local, not global.

PredefinedColorsMap isn't anything else than a function, you pass a PredefinedColor to it, and you get how you must render the colors according to this map. If you want to change how to represent the colors, you just use another map.

I don't see strong reasons to change this.